### DISCUSSION TITLE
Method of Manufactured Solutions (MMS): running example results in error

### DISCUSSION THREAD

**Discussion by klean700c** [idaholab/moose#20356] [Q&A Tools] [answered]
Dear MOOSE community,

I tried to run the documented example from https://mooseframework.inl.gov/python/mms.html .

```
#!/usr/bin/env python3

import mms
fs,ss = mms.evaluate('-div(grad(u))', 'sin(2*pi*x)*sin(2*pi*y)')
mms.print_fparser(fs)

mms.print_hit(fs, 'force')
mms.print_hit(ss, 'exact')
```
I ran the program by

```
(base) jan@Jan-ThinkPad:~$ mamba activate moose
(moose) jan@Jan-ThinkPad:~/projects$ python3 mms_exact.py
```

Unfortunatelly, it returns the following error message
```
Traceback (most recent call last):
  File "/home/jan/projects/mms_exact.py", line 11, in <module>
    import mms
ModuleNotFoundError: No module named 'mms'
```
I installed sympy before I ran the code.
Can you explain me how to implement the mms module?
I´m a new user to MOOSE and Ubuntu 18.04.5. Perhaps I did some obvious error, that was not that obvious to me. Although, compiling and testing MOOSE went successful.

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

mms is one of our packages. 
You just need to tell Python where to look for that module. We by default dont modify your PYTHONPATH

so 
``` export PYTHONPATH=$PYTHONPATH:/home/jan/projects/moose/python```

Guillaume

**Reply by hugary1995** [reply]
Is it possible to let the conda environment modify PYTHONPATH for us? I've run into this issue several times already.

**Reply by GiudGiud** [reply]
it's certainly possible if we know where people installed moose.
@milljm what do you think?
